app/controllers/submissions_controller.rb

```diff
@@ -473,23 +499,36 @@ def turnitin_report
       end
     end
   end
+  private :plagiarism_report
 
-  def resubmit_to_turnitin
+  def resubmit_to_plagiarism(type)
     return render(:nothing => true, :status => 400) unless params_are_integers?(:assignment_id, :submission_id)
 
     if authorized_action(@context, @current_user, [:manage_grades, :view_all_grades])
       @assignment = @context.assignments.active.find(params[:assignment_id])
       @submission = @assignment.submissions.where(user_id: params[:submission_id]).first
-      @submission.resubmit_to_turnitin
+
+      if type == 'vericite'
+        # VeriCite
+        @submission.resubmit_to_vericite
+        messageKey = "resubmitted_to_vericite"
+        messageDefault = "Successfully resubmitted to VeriCite."
+      else
+        # turnitin
+        @submission.resubmit_to_turnitin
+        messageKey = "resubmitted_to_turnitin"
+        messageDefault = "Successfully resubmitted to turnitin."
+      end
       respond_to do |format|
         format.html {
-          flash[:notice] = t('resubmitted_to_turnitin', "Successfully resubmitted to turnitin.")
+          flash[:notice] = t(messageKey, messageDefault)
```

Comment:
This doesn't work for I18n, since we do static analysis to extract the strings to be translated... `t()` calls always need a literal string. Additionally, specifying a key is not actually necessary anymore, so you could do something like:

```ruby
if type == 'vericite'
  @submission.resubmit_to_vericite
  message = t("Successfully resubmitted to VeriCite.")
else
  @submission.resubmit_to_turnitin
  message = t("Successfully resubmitted to turnitin.")
end
respond_to do |format|
  format.html {
    flash[:notice] = message
```